src/components/timecard/edit-shift-timecard/EditShiftTimecard.jsx

```diff
@@ -2,19 +2,39 @@ import { PropTypes } from 'prop-types';
 import { useState } from 'react';
 import { Link } from 'react-router-dom';
 import EditShiftHours from '../edit-shift-hours/EditShiftHours';
+import { deleteTimeEntry } from '../../../api/services/timecardService';
+import { UrlSearchParamBuilder } from '../../../utils/api-utils/UrlSearchParamBuilder';
+import { useTimecardContext } from '../../../context/TimecardContext';
+import { deepClone } from '../../../utils/common-utils/common-utils';
 
 const EditShiftTimecard = ({ timeEntry, timeEntriesIndex }) => {
   const toggleEditShiftHours = (event) => {
     event.preventDefault();
     setShowEditShiftHours(!showEditShiftHours);
   };
 
-  const timeEntryIsEmpty =
+  const { timeEntries, setTimeEntries } = useTimecardContext();
+
+  const timeEntryExists =
     'startTime' in timeEntry && timeEntry.startTime !== '';
   const [showEditShiftHours, setShowEditShiftHours] = useState(
-    !timeEntryIsEmpty
+    !timeEntryExists
   );
 
+  const clickRemoveShiftButton = async (event) => {
+    event.preventDefault();
+    const params = new UrlSearchParamBuilder()
+      .setTenantId('00000000-0000-0000-0000-000000000000')
+      .getUrlSearchParams();
+    const response = await deleteTimeEntry(timeEntry.timeEntryId, params);
+
+    if (response.status === 200) {
```

Comment:
I think we need some tests to cover this logic here of setting the new time entries that excludes the removed entry.

Reply:
We've added a new test for this now

Reply:
If it's not too much trouble could we add a similar test to the one you've added / just replace it to cover the situation where you have more than one entry e.g 2 and then you expect `mockSetTimeEntries` to have been called with 1 less entry e.g 1.
Only mentioning as we're using splicing and indexes so it just gives a bit more reassurance. 
If this ends up holding it back then no worries.

Reply:
Added another one now